Shared/IISFunctions/Get-IISModules.ps1

```diff
@@ -0,0 +1,159 @@
+                            }
+
+                            $mInfoObject.Path = $moduleFilePath
+
+                            try {
+                                Write-Verbose "Querying file signing information"
+                                $signature = Get-AuthenticodeSignature -FilePath $moduleFilePath -ErrorAction Stop
+                                # Signature Status Enum Values:
+                                # <0> Valid, <1> UnknownError, <2> NotSigned, <3> HashMismatch,
+                                # <4> NotTrusted, <5> NotSupportedFileFormat, <6> Incompatible,
+                                # https://docs.microsoft.com/dotnet/api/system.management.automation.signaturestatus
+                                if (($signature.Status -ne 1) -and
+                                    ($signature.Status -ne 2) -and
+                                    ($signature.Status -ne 5) -and
+                                    ($signature.Status -ne 6)) {
+                                    Write-Verbose "Signature information found. Status: $($signature.Status)"
+
+                                    $mInfoObject.Signed = $true
+                                    $mDetails = [PSCustomObject]@{
+                                        Signer            = $null
+                                        SignatureStatus   = $signature.Status
+                                        IsMicrosoftSigned = $false
+                                    }
+
+                                    if ($null -ne $signature.SignerCertificate.Subject) {
+                                        Write-Verbose "Signer information found. Subject: $($signature.SignerCertificate.Subject)"
+                                        $mDetails.Signer = $signature.SignerCertificate.Subject.ToString()
+                                        $mDetails.IsMicrosoftSigned = $signature.SignerCertificate.Subject -cmatch "O=Microsoft Corporation, L=Redmond, S=Washington"
+                                    }
+
+                                    $mInfoObject.SignatureDetails = $mDetails
+                                }
+
+                                $iisModulesList.Add($mInfoObject)
+                            } catch {
+                                Write-Verbose "Unable to validate file signing information"
+                                if ($null -ne $CatchActionFunction) {
+                                    & $CatchActionFunction
+                                }
+                            }
+                        }
+                    } else {
+                        Write-Verbose "No modules are loaded by IIS"
+                    }
+                } catch {
+                    Write-Verbose "Failed to process global module information. $_"
+                    if ($null -ne $CatchActionFunction) {
+                        & $CatchActionFunction
+                    }
```

Comment:
Use shared function `Invoke-CatchActionError`

Reply:
`Get-IISModules` was designed as a shared function that can be used by multiple projects. So I decided to not add `Invoke-CatchActionError` as a dependency. Instead, it's possible to pass a catch action function if you want to use one, with the `CatchActionFunction` parameter.

Reply:
Please reopen if you want to have this changed.

Reply:
`Invoke-CatchActionError` is a `Shared` function already that is doing the same thing you currently have. 

![image](https://user-images.githubusercontent.com/22776718/182903885-2a854fa0-dd72-4a07-83ae-553ff5e3fe2b.png)

Reply:
See `Invoke-ScriptBlockHandler` for an example.

Reply:
Ah I see. Okay, will make the change.

Reply:
Addressed.